nvsvocprez/app.py

```diff
@@ -1536,19 +1536,26 @@ def __init__(self, request):
             "sdo": sdo,
         }
 
-        def _is_collection_puv():
-            collection_uri = self.instance_uri.split("/current/")[0] + "/current/"
-            for collection in cache_return(collections_or_conceptschemes="collections"):
-                if collection["uri"]["value"] == collection_uri:
-                    if (
-                        collection.get("conforms_to")
-                        and "https://w3id.org/env/puv" in collection["conforms_to"]["value"]
-                    ):
-                        return True
-            return False
-
-        if _is_collection_puv():
-            concept_profiles["puv"] = puv
+        altprof_data = get_alt_profiles()
+        collection_uri = self.instance_uri.split("/current/")[0] + "/current/"
+        for collection in cache_return(collections_or_conceptschemes="collections"):
+            if collection["uri"]["value"] == collection_uri:
+                if collection.get("conforms_to"):
+                    conforms_to = collection["conforms_to"]["value"].split(",")
+                    for profile in conforms_to:
+                        alt_data = altprof_data.get(profile, None)
+                        if not alt_data or alt_data["token"] in concept_profiles:
+                            continue
+                        concept_profiles[alt_data["token"]] = Profile(
+                            uri=alt_data["url"],
+                            id=alt_data["token"],
+                            label=alt_data["name"],
+                            comment=alt_data["vocprezdesc"],
+                            mediatypes=["text/html"] + RDF_MEDIATYPES,
+                            default_mediatype="text/html",
+                            languages=["en"],
+                            default_language="en",
+                        )
```

Comment:
I have a method get_alt_profile_objects() in my branch to generate the Profile objects for each alternate profile. May be able to be reused here 🤔